opencensus/log/__init__.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import namedtuple
+from copy import copy
+import logging
+
+from opencensus.trace import execution_context
+
+
+_meta_logger = logging.getLogger(__name__)
+
+TRACE_ID_KEY = 'traceId'
+SPAN_ID_KEY = 'spanId'
+SAMPLING_DECISION_KEY = 'traceSampled'
+
+LogAttrs = namedtuple('LogAttrs', ['trace_id', 'span_id', 'sampling_decision'])
+ATTR_DEFAULTS = LogAttrs('', '', False)
+
+
+def get_log_attrs():
+    """Get logging attributes from the opencensus context.
+
+    :rtype: :class:`LogAttrs`
+    :return: The current span's trace ID, span ID, and sampling decision.
+    """
+    try:
+        tracer = execution_context.get_opencensus_tracer()
+        if tracer is None:
+            raise RuntimeError
+    except Exception:  # noqa
+        _meta_logger.error("Failed to get opencensus tracer")
+        return ATTR_DEFAULTS
+
+    try:
+        trace_id = tracer.span_context.trace_id
+        if trace_id is None:
+            trace_id = ATTR_DEFAULTS.trace_id
+    except Exception:  # noqa
+        _meta_logger.error("Failed to get opencensus trace ID")
+        trace_id = ATTR_DEFAULTS.trace_id
+
+    try:
+        span_id = tracer.span_context.span_id
+        if span_id is None:
+            span_id = ATTR_DEFAULTS.span_id
+    except Exception:  # noqa
+        _meta_logger.error("Failed to get opencensus span ID")
+        span_id = ATTR_DEFAULTS.span_id
+
+    try:
+        sampling_decision = tracer.span_context.trace_options.get_enabled
```

Comment:
Curious, why do we want to be specific about trace_options.get_enabled instead of the whole trace_options (e.g. if there are other flags in the future)?

Reply:
This is just to follow the spec in this PR. I think there's good reason to include (a configurable subset of) the trace options, tracestate, tags, etc. in the future.

Reply:
We also need to check `get_enabled` here because of the roundabout way we're handling sampling decisions. We'll need to change this after fixing #481.

Reply:
Got it. Thanks for the background.